sklearnex/basic_statistics/basic_statistics.py

```diff
@@ -14,4 +14,130 @@
+        sum_squares_centered : ndarray of shape (n_features,)
+            Centered sum of squares for each feature over all samples.
+        second_order_raw_moment : ndarray of shape (n_features,)
+            Second order moment of each feature over all samples.
+    """
+
+    def __init__(self, result_options="all"):
+        self.options = result_options
+
+    _onedal_basic_statistics = staticmethod(onedal_BasicStatistics)
+
+    def _save_attributes(self):
+        assert hasattr(self, "_onedal_estimator")
+
+        if self.options == "all":
+            result_options = onedal_BasicStatistics.get_all_result_options()
+        else:
+            result_options = self.options
+
+        if isinstance(result_options, str):
+            setattr(self, result_options, getattr(self._onedal_estimator, result_options))
+        elif isinstance(result_options, list):
+            for option in result_options:
+                setattr(self, option, getattr(self._onedal_estimator, option))
+
+    def _onedal_supported(self, method_name, *data):
+        patching_status = PatchingConditionsChain(
+            f"sklearnex.basic_statistics.{self.__class__.__name__}.{method_name}"
+        )
+        return patching_status
+
+    _onedal_cpu_supported = _onedal_supported
+    _onedal_gpu_supported = _onedal_supported
+
+    def _onedal_fit(self, X, sample_weight=None, queue=None):
+        if sklearn_check_version("1.0"):
+            X = self._validate_data(X, dtype=[np.float64, np.float32])
+        else:
+            X = check_array(
+                X,
+                dtype=[np.float64, np.float32],
+            )
+
+        if sample_weight is not None:
+            sample_weight = _check_sample_weight(sample_weight, X)
+
+        onedal_params = {
+            "result_options": self.options,
+        }
+
+        if not hasattr(self, "_onedal_estimator"):
+            self._onedal_estimator = self._onedal_basic_statistics(**onedal_params)
+        self._onedal_estimator.fit(X, sample_weight, queue)
+        self._save_attributes()
+
+    def fit(self, X, y=None, *, sample_weight=None):
+        """Compute statistics with X, using minibatches of size batch_size.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Data for compute, where `n_samples` is the number of samples and
+            `n_features` is the number of features.
+
+        y : Ignored
+            Not used, present for API consistency by convention.
+
+        sample_weight : array-like of shape (n_samples,), default=None
+            Weights for compute weighted statistics, where `n_samples` is the number of samples.
+
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        dispatch(
+            self,
+            "fit",
+            {
+                "onedal": self.__class__._onedal_fit,
+                "sklearn": None,
```

Comment:
it will fail. there are few such algos in our repo. probably it is better to resolve (add exception raising, I guess) in a separate pr. we have jira ticket about that.